code/system/application/perception/src/detectlane/detectlane.cpp

```diff
@@ -94,437 +130,165 @@ void DetectLane::tearDown()
-          std::cout << m_roi[0]<<" x " << m_roi[1]<<" x " << m_roi[2] <<" x " << m_roi[3] << std::endl;
-          break;
-        case 'k':
-          {
-            odcore::base::Lock l(m_mtx);
-            m_roi[3] = m_roi[3] - sgn*5;
-            if(m_roi[3] > m_screenSize[1]) {
-              m_roi[3] = m_screenSize[1];
-            } else if(m_roi[1] + m_roi[3] > m_screenSize[1]) {
-              m_roi[3] = m_screenSize[1] - m_roi[1];
-            }
-          }
-          m_visualMemory.clear();
-          std::cout << m_roi[0]<<" x " << m_roi[1]<<" x " << m_roi[2] <<" x " << m_roi[3] << std::endl;
-          break;
-        case'y':
-          m_memThreshold += 0.1;
-          std::cout << "Mem: " << m_memThreshold << std::endl;
-          break;
-        case'h':
-          m_memThreshold -= 0.1;
-          if (m_memThreshold < 0.1) {
-            m_memThreshold = 0.1;
-          }
-          std::cout << "Mem: " << m_memThreshold << std::endl;
-          break;
-        case't':
-          m_cannyThreshold += 5;
-          std::cout << "Canny: " << m_cannyThreshold << std::endl;
-          break;
-        case'g':
-          m_cannyThreshold -= 5;
-          if (m_cannyThreshold > 255) {
-            m_cannyThreshold = 0;
-          }
-          std::cout << "Canny: " << m_cannyThreshold << std::endl;
-          break;
-        case'r':
-          m_houghThreshold += 5;
-          std::cout << "Hough: " << m_houghThreshold << std::endl;
-          break;
-        case'f':
-          m_houghThreshold -= 5;
-          if (m_houghThreshold < 10) {
-            m_houghThreshold = 15;
-          }
-          std::cout << "Hough: " << m_houghThreshold << std::endl;
-          break;
-        case'w':
-          m_upperLaneLimit -= 5;
-          std::cout << "UpperLaneLimit: " << m_upperLaneLimit << std::endl;
-          break;
-        case's':
-          m_upperLaneLimit += 5;
-          std::cout << "UpperLaneLimit: " << m_upperLaneLimit << std::endl;
-          break;
-        case'q':
-          m_lowerLaneLimit -= 5;
-          std::cout << "LowerLaneLimit: " << m_lowerLaneLimit << std::endl;
-          break;
-        case'a':
-          m_lowerLaneLimit += 5;
-          std::cout << "LowerLaneLimit: " << m_lowerLaneLimit << std::endl;
-        default:
-          break;
-      }
-    }
-  }
-  return odcore::data::dmcp::ModuleExitCodeMessage::OKAY;
-}
-
+// odcore::data::dmcp::ModuleExitCodeMessage::ModuleExitCode DetectLane::body()
+// {
+//   while (getModuleStateAndWaitForRemainingTimeInTimeslice() ==
+//   odcore::data::dmcp::ModuleStateMessage::RUNNING)
+//   {}
+//   return odcore::data::dmcp::ModuleExitCodeMessage::OKAY;
+// }
 
 /**
  * Receives images from vision source and detect lanes.
  * Sends drivalble lanes objects.
  */
 void DetectLane::nextContainer(odcore::data::Container &a_c)
 {
-  if (!m_initialized || a_c.getDataType() != odcore::data::image::SharedImage::ID()) {
+  if (!m_initialized) {
     return;
```

Comment:
Afaik, the nextContainer function is an independent thread which could invoke member function which uses member variables before setUp() has been executed properly causing a segmentation fault. This might be fixed already?